lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -958,40 +966,13 @@ evaluateTransactionBalance era pp utxo txBody' =
 -- Module-internal helpers
 --------------------------------------------------------------------------------
 
-withShelleyEraTxBodyConstraint
-    :: RecentEra era
-    -> ((Babbage.ShelleyEraTxBody (ShelleyLedgerEra era)) => a)
-    -> a
-withShelleyEraTxBodyConstraint era a = case era of
-    RecentEraAlonzo -> a
-    RecentEraBabbage -> a
-    RecentEraConway -> a
-
-withStandardCryptoConstraint
-    :: RecentEra era
-    -> ((Crypto (Cardano.ShelleyLedgerEra era) ~ StandardCrypto) => a)
-    -> a
-withStandardCryptoConstraint era a = case era of
-    RecentEraAlonzo -> a
-    RecentEraBabbage -> a
-    RecentEraConway -> a
-
-withCLIConstraint
-    :: RecentEra era
-    -> (CLI (ShelleyLedgerEra era) => a)
-    -> a
-withCLIConstraint era a = case era of
-    RecentEraAlonzo -> a
-    RecentEraBabbage -> a
-    RecentEraConway -> a
-
-withAlonzoScriptConstraint
+-- | Bring constraints related to 'IsRecentEra' into scope from a value-level
+-- 'RecentEra'.
+withConstraints
     :: RecentEra era
-    -> ( Core.Script (Cardano.ShelleyLedgerEra era)
-         ~ AlonzoScript (Cardano.ShelleyLedgerEra era)
-        => a)
+    -> (IsRecentEra era => a)
```

Comment:
Couldn't get rid of these helpers completely as we still take `RecentEra` on the value-level. Need for value-level `RecentEra` should be reducible with ADP-2353 though.